enhancements/security/openstack-self-signed-certificates.md

```diff
@@ -0,0 +1,137 @@
+
+The `cluster-image-registry-operator` (CIRO) and the `image-registry` also need to communicate with OpenStack in order to use swift. CIRO is allowed to read configmaps from the `openshift-config` namespace, so it simple uses an openshift listener to get the latest certificate from the `cloud-provider-config` configmap, and can create a custom http transport that trusts the certificate for authenticating with OpenStack.
+
+The `machine-controllers` and more specifically, `cluster-api-provider-openstack` (CAPO), will need to authenticate with openstack any time it needs to manage the current machine pool. In order to keep `openshift-config/cloud-provider-config` as the central point that services draw the CA certificate from, CAPO is given RBAC access to `GET` configmaps named `cloud-provider-config` from the `openshift-config` namespace. This allows the kubernetes client in CAPO to fetch the latest CA certificate, and craft a custom http transport that trusts it when authenticating with OpenStack.
+
+Lastly, Kuryr needs to communicate with OpenStack networking services as part of its core functionality. Following the common design pattern, kuryr GETs the latest version of `openshift-config/cloud-provider-config` and builds a custom http transport with the CA cert when authenticating with OpenStack.
+
+In total, these changes enable the distribution, and trust of OpenStack CA certificates in an OpenShift cluster both during and after install.
+
+## Motivation
+
+Numerous customer complaints have resulted in this issue being highly escalated. This prevents a number of customers from installing OpenShift 4 on their OpenStack clusters.
+
+### Goals
+
+1. To enable the OpenShift 4.4 installer to succesfully deploy a cluster on an OpenStack cloud that uses self signed certificates for auth.
+2. Openshift 4.4 can succesfully and stabily run on OpenStack clusters that use self signed certificates for auth.
+3. Backport to 4.3
+
+## Proposal
+
+### Risks and Mitigations
+
+1. It is possible to give services authorization/credentials that they dont need and shouldnt have, and this may pose a security risk.
+
+> To prevent this, we make sure that only services that need the CA cert get access to it, and the cert we pass them only has the CA cert in it.
```

Comment:
This is just the serve cert, right?  There's no way to leverage that for privilege escalation is there, even if it was world-readable?  If this was about client TLS certs/keys, that would be different, but as far as I can tell client-side auth is not covered in this enhancement.